plugins/experimental/stek_share/stek_share.cc

```diff
@@ -422,7 +422,7 @@ stek_updater(void *arg)
   ::pthread_setcancelstate(PTHREAD_CANCEL_ENABLE, nullptr);
   ::pthread_setcanceltype(PTHREAD_CANCEL_ASYNCHRONOUS, nullptr);
 
-  Dbg(dbg_ctl, "Starting STEK updater thread: %lu", ::pthread_self());
+  Dbg(dbg_ctl, "Starting STEK updater thread");
```

Comment:
Considering the variety of types for pthread_self, and that @duke8253 says this information is not needed, I'd still lean toward just removing the use of it in the debug log. It's not obvious to me that arbitrarily casting from the possible variety of types to an int type is safe.

Reply:
I'll merge as is to alleviate the warning. If we want a more specific log message, I'd suggest implementing a separate function that returns an unsigned long from the variety of types. It can inspect `__APPLE__` and handle that and possibly other situations too (not sure offhand what freebsd does, for instance).